_posts/2020-11-09-pcr-primer-calculator-using-r.md

```diff
@@ -0,0 +1,295 @@
+    - capitalized all letters in the primer sequence (in column E) using stringr
+    - lowercase all letters in the restriction site sequences (in column D) using stringr
+    - capitalize all letters in the leading strand sequence using stringr
+    - store the complete PCR primer (leading strand, restriction site, and primer sequence) together to a variable
+    - store the complete PCR primer with "5'" and "3'" wrapped around the sequence to another variable
+    - count the number of A's, T's, G's, C's, etc. in the annealing portion of the sequence to individual variables using stringr
+    - if the sequence has ambiguous bases:
+      - [Yes] calculate the minimum and maximum melting temperature
+      - [No] calculate the melting temperature
+    - create variables to hold the gc-percentage and annealing primer length
+    - create a data frame using the variables we created
+4. Export the results of the calculator to a folder as a new csv file
+  - use readr to write the data frame to a csv file.
+    - save the file using the row number and primer name
+  - close the For Loop
+
+Lets take a look at the expected input file for our script.  
+![input file layout]({{ site.url }}/assets/pcr-primer-calculator-using-r-input-file.jpg)
+
+For each row with sequencing information, we want to process and output the results of PCR primer.  
+![output folder layout]({{ site.url }}/assets/pcr-primer-calculator-using-r-output-folder.jpg)
+
+Make a note about each column. Reference the column inside the code, so that we can denote each code block and what field that code relates to.  
+![output file results]({{ site.url }}/assets/pcr-primer-calculator-using-r-output-file-results.jpg)
+
+Each output file will contain the entire PCR primer sequence (leading strand, restriction site and primer sequence), a count of each nucleotide of the *annealing* portion of the primer sequence, and the melting temperature of the primer sequence. Furthermore, we need to add in a minimum and maximum melting temperature section to the output file. I will explain why we need a minimum and maximum once we begin coding the nucleotide count code.
+
+-------------------------------------------------------------------------------
+## Creating our script in R
+#### Section 1: Adding libraries to the R script  
+
+Libraries were created to make coding easier for everyone. A library inside code is synonymous to a literal library. They are full of *books*. Each book you request from the library helps you accomplish a task. Imagine yourself as a computer, your human overlord demands that you take the integral of an infinite series that is both continuous and decreasing in a positive manner and needs to know if the function converges or diverges. What are you going to do? You will go to the library and request a Calculus textbook by Ron Larson. Once you have the book, you can tell your master the area under the curve of the infinite series is finite and will converge.
```

Comment:
R nerds on the internet might call you on this. They are actually called packages, and are loaded with the `library()` function. So you should probably revise this appropriately. It's a common mistake but worth having the post be technically correct. You can still use the library metaphor, just refer to the function and not the package itself.